File: RQ1/CodeForces-Automatic/318A.c
#include <stdio.h>
#include <assert.h>

#define ll long long

int main() {
    ll n, k;
    scanf("%lld %lld", &n, &k);
    
    assert(n >= 1 && k >= 1 && k <= n); // PRE: 1 ≤ k ≤ n ≤ 10^12
    
    ll result;
    if (k <= n / 2)
        result = 2 * k - 1;
    else {
        if (n & 1) {
            if (k == n / 2 + 1)
                result = n;
            else
                result = 2 * (k - n / 2 - 1);
        } else
            result = 2 * (k - n / 2);
    }
    
    printf("%lld\n", result);
    
    assert(result >= 1 && result <= n); // POST: Output number is in the range of 1 to n
    return 0;
}

File: RQ1/CodeForces-Automatic/318A.py
def main():
    # PRE:
    # Input Format:
    # The only line of input contains integers n and k (1 ≤ k ≤ n ≤ 1012).Please, do not use the %lld specifier to read or write 64-bit integers in C++. It is preferred to use the cin, cout streams or the %I64d specifier.
    # POST:
    # Program Description:
    # Being a nonconformist, Volodya is displeased with the current state of things, particularly with the order of natural numbers (natural number is positive integer number). He is determined to rearrange them. But there are too many natural numbers, so Volodya decided to start with the first n. He writes down the following sequence of numbers: firstly all odd integers from 1 to n (in ascending order), then all even integers from 1 to n (also in ascending order). Help our hero to find out which number will stand at the position number k.
    # Output Format:
    # Print the number that will stand at the position number k after Volodya's manipulations.
    from math import  floor
    n, k = map(int, input().split())
    
    odds = range(1, n+1, 2)
    evens = range(2, n+1, 2)
    
    try:
          print(odds[k-1])
    except IndexError:
          try:
                idx = k - len(odds)
                print(evens[idx-1])
          except OverflowError:
                try:
                      idx = k - floor(n/2)
                      print(evens[idx-1])
                except IndexError:
                      print(evens[idx-2])
